library/ssl_tls13_client.c

```diff
@@ -2617,30 +2617,33 @@ static int ssl_encrypted_extensions_parse( mbedtls_ssl_context* ssl,
                                            unsigned char* buf,
                                            size_t buflen )
 {
-    int ret=0;
+    int ret = 0;
     size_t ext_len;
     unsigned char *ext;
 
-    /* TODO: Add bounds checks! Only then remove the next line. */
-    ((void) buflen );
+    if( buflen < 2 )
+    {
+        MBEDTLS_SSL_DEBUG_MSG( 1, ( "Extension buffer length too short - bad encrypted extensions message" ) );
+        return( MBEDTLS_ERR_SSL_BAD_HS_ENCRYPTED_EXTENSIONS );
```

Comment:
It is not clear to me whether we want to keep using such fine-grained per-message error codes. I have raised this with the team.